orc8r/cloud/go/obsidian/swagger/spec_servicer_test.go

```diff
@@ -15,43 +15,47 @@ package swagger_test
 
 import (
 	"context"
+	"errors"
 	"io/ioutil"
 	"os"
 	"path/filepath"
 	"testing"
 
 	"magma/orc8r/cloud/go/obsidian/swagger"
+	"magma/orc8r/cloud/go/obsidian/swagger/mocks"
 	"magma/orc8r/cloud/go/obsidian/swagger/protos"
 
 	"github.com/stretchr/testify/assert"
 )
 
 func TestSpecServicer_NewSpecServicerFromFile(t *testing.T) {
+	// Missing default dir / spec files should not panic.
+	servicer := swagger.NewSpecServicerFromFile("foo")
+	assert.NotNil(t, servicer)
+}
+
+func TestSpecServicer_NewSpecServicerFromSpecs(t *testing.T) {
+	d, err := ioutil.TempDir("", "")
+	assert.NoError(t, err)
+	defer os.RemoveAll(d)
+
+	partialDir := filepath.Join(d, "partial")
+	standaloneDir := filepath.Join(d, "standalone")
+	assert.NoError(t, os.MkdirAll(partialDir, os.ModePerm))
+	assert.NoError(t, os.MkdirAll(standaloneDir, os.ModePerm))
+
 	testFile := "test_spec_servicer.swagger.v1.yml"
 	testPartialFileContents := "test partial yaml spec"
 	testStandaloneFileContents := "test standalone yaml spec"
-	tmpDir := "/etc/magma/swagger/specs"
-	partialDir := "/etc/magma/swagger/specs/partial"
-	standaloneDir := "/etc/magma/swagger/specs/standalone"
-
-	os.RemoveAll(tmpDir)
-	defer os.RemoveAll(tmpDir)
-
-	err := os.MkdirAll(partialDir, os.ModePerm)
-	assert.NoError(t, err)
-	err = os.MkdirAll(standaloneDir, os.ModePerm)
-	assert.NoError(t, err)
 
 	tmpPartialSpecPath := filepath.Join(partialDir, testFile)
-	err = ioutil.WriteFile(tmpPartialSpecPath, []byte(testPartialFileContents), 0644)
-	assert.NoError(t, err)
+	assert.NoError(t, ioutil.WriteFile(tmpPartialSpecPath, []byte(testPartialFileContents), 0644))
```

Comment:
The reason I don't like doing it like this is it "reads" in a surprising way -- you start by reading that this is an assert line, which you think is nullipotent, but then come to find out, later in the line, that the line actually has a side effect / is taking an action

To me, it reads more naturally to follow this pattern

```suggestion
	err = ioutil.WriteFile(tmpPartialSpecPath, []byte(testPartialFileContents), 0644)
	assert.NoError(t, err)
```